src/Sulu/Bundle/MediaBundle/Resources/public/js/components/media-selection/main.js

```diff
@@ -203,6 +211,95 @@ define(['sulumedia/collection/collections', 'sulumedia/model/collection'], funct
             this.sandbox.on('sulu.grid-group.' + this.options.instanceName + '.record-deselected', function(id) {
                 this.sandbox.emit(RECORD_DESELECTED.call(this), id);
             }.bind(this));
+
+            this.sandbox.on('husky.data-navigation.' + this.options.instanceName + '.select', function(collection) {
+                var collectionId, 
+                    collectionTitle = this.sandbox.translate('media-selection.overlay.all-images');
+
+                if (collection) {
+                    collectionId = collection.id;
+                    collectionTitle = collection.title;
+
+                    this.sandbox.emit('husky.toolbar.media-selection-ovelay.' + this.options.instanceName + '.item.enable', 'add');
+                    this.sandbox.emit('husky.dropzone.media-selection-ovelay.' + this.options.instanceName + '.enable');
+                } else {
+                    this.sandbox.emit('husky.toolbar.media-selection-ovelay.' + this.options.instanceName + '.item.disable', 'add');
+                    this.sandbox.emit('husky.dropzone.media-selection-ovelay.' + this.options.instanceName + '.disable');
+                }
+
+                this.sandbox.emit('husky.datagrid.media-selection-ovelay.' + this.options.instanceName + '.url.update', { collection: collectionId });
+                changeUploadCollection.call(this, collectionId);
+                this.$el.find('.media-selection-overlay-content-title').html(collectionTitle);
```

Comment:
Usually we try to keep the listeners in the `bindCustomEvents` method quite small...